### PR TITLE
Back-end setup

### PR DESCRIPTION
**Pull request by quinnouyang**
Spotify API examples, refreshed `.gitgnore`.

### CODE REVIEW

File: .github/workflows/jest-test.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - 'server/**.ts'
+      - '!server/node_modules/**'
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  run-test:
+    name: Run test
+    runs-on: ubuntu-latest
```
Comment (jasyip) [inline .github/workflows/jest-test.yml:15]: consider other OS? copy my usage of GitHub Actions matrix

**Review by jasyip — APPROVED** [review]

File: .github/workflows/jest-test.yml (hunk 1 above)
Comment (quinnouyang) [reply .github/workflows/jest-test.yml:15]: As long as it's consistent and works for deployment, fine by me. None of our local machines replicate this exactly but we could containerize.

**Comment by quinnouyang** [comment]
@prathyushreddylpr Could you look through this? Sorry for the mess:)

**Review by Heaper327 — CHANGES_REQUESTED** [review]
Everything looks good, but we should remove dependency directories like node_modules before merging

**Comment by quinnouyang** [comment]
Definitely, I've tried resetting the git cache after adding those to `.gitignore`. Might have to manually remove these for now

**Comment by Heaper327** [comment]
I have already removed the dependency files on remote, and resetting the git cache fixed my local staging area:
```
git rm -r --cached .
git add .
```
Could you take a look at the files again? I think this branch has done what it needed to do and could be merged now.

**Review by Heaper327 — APPROVED** [review]

**Pull request merged**